src/utils/sentry/httpclient.ts

```diff
@@ -0,0 +1,486 @@
+// forked from https://github.com/getsentry/sentry-javascript/blob/55d41f7c20b8cfcafda9bbeef75d551c1e7e22d7/packages/browser/src/integrations/httpclient.ts#L40
+import { captureEvent, defineIntegration, getClient, isSentryRequestUrl } from '@sentry/core'
+import type {
+  Client,
+  Event as SentryEvent,
+  IntegrationFn,
+  SentryWrappedXMLHttpRequest,
+} from '@sentry/types'
+import {
+  addExceptionMechanism,
+  addFetchInstrumentationHandler,
+  GLOBAL_OBJ,
+  logger,
+  supportsNativeFetch,
+} from '@sentry/utils'
+import { addXhrInstrumentationHandler, SENTRY_XHR_DATA_KEY } from '@sentry-internal/browser-utils'
+
+export type HttpStatusCodeRange = [number, number] | number
+export type HttpRequestTarget = string | RegExp
+
+const INTEGRATION_NAME = 'HttpClient'
+
+interface HttpClientOptions {
+  /**
+   * HTTP status codes that should be considered failed.
+   * This array can contain tuples of `[begin, end]` (both inclusive),
+   * single status codes, or a combinations of both
+   *
+   * Example: [[500, 505], 507]
+   * Default: [[500, 599]]
+   */
+  failedRequestStatusCodes: HttpStatusCodeRange[]
+
+  /**
+   * An optional list of targets to allow tracking of failed requests.
+   * This array can contain strings or regular expressions.
+   * If omitted, no filtering by requests whitelist will be applied.
+   *
+   * Example: ['http://localhost', /api\/.*\/]
+   * Default: [/.*\/]
+   */
+  failedRequestTargets?: HttpRequestTarget[]
+  /**
+   * An optional list of URLs to deny tracking of failed requests.
+   * This array can contain strings, not regular expressions.
+   * If omitted, no filtering by requests blacklist will be applied.
+   *
+   * Example: ['snapshot.org', 'alchemy']
+   */
+  denyUrls?: string[]
+}
+
+const _httpClientIntegration = ((options: Partial<HttpClientOptions> = {}) => {
+  const _options: HttpClientOptions = {
+    failedRequestStatusCodes: [[500, 599]],
+    failedRequestTargets: [/.*/],
+    ...options,
+  }
+
+  return {
+    name: INTEGRATION_NAME,
+    setupOnce() {},
+    setup(client): void {
+      _wrapFetch(client, _options)
+      _wrapXHR(client, _options)
+    },
+  }
+}) as IntegrationFn
+
+/**
+ * Create events for failed client side HTTP requests.
+ */
+export const httpClientIntegration = defineIntegration(_httpClientIntegration)
+
+/**
+ * Interceptor function for fetch requests
+ *
+ * @param requestInfo The Fetch API request info
+ * @param response The Fetch API response
+ * @param requestInit The request init object
+ */
+function _fetchResponseHandler(
```

Comment:
Can we use ES6 arrow functions for consistency with the rest of web?

Reply:
Ditto diffing purposes trying to change as little as possible from the original implementation